paddle/fluid/framework/ir/graph_pattern_detector.cc

```diff
@@ -517,6 +517,37 @@ bool VarLinksFromOp(Node* node, const std::string& op_type) {
   return false;
 }
 
+PDNode* patterns::ConvReLU(PDPattern* pattern, const std::string& name_scope,
+                           PDNode* conv_input) {
+  // Create Operators
+  auto* conv_op = pattern->NewNode(name_scope, "conv")->assert_is_op("conv2d");
+  auto* relu_op = pattern->NewNode(name_scope, "relu")->assert_is_op("relu");
+  // Create variables
+  // Filter
+  auto* conv_weight_var = pattern->NewNode(name_scope, "conv_weight")
+                              ->AsInput()
+                              ->assert_is_persistable_var()
+                              ->assert_is_op_nth_input("conv2d", "Filter", 0);
+  // Bias
+  auto* conv_bias_var = pattern->NewNode(name_scope, "conv_bias")
+                            ->AsInput()
+                            ->assert_is_persistable_var()
+                            ->assert_is_op_nth_input("conv2d", "Bias", 0);
+  // intermediate variable, will be removed in the IR after fuse.
+  auto* conv_out_var = pattern->NewNode(name_scope, "conv_out")
+                           ->AsIntermediate()
+                           ->assert_is_only_output_of_op("conv2d")
+                           ->assert_is_op_input("relu");
+  // output
+  auto* relu_out_var = pattern->NewNode(name_scope, "relu_out")
+                           ->AsOutput()
+                           ->assert_is_op_output("relu");
+
+  conv_op->LinksFrom({conv_input, conv_weight_var, conv_bias_var})
+      .LinksTo({conv_out_var});
+  relu_op->LinksFrom({conv_out_var}).LinksTo({relu_out_var});
+  return relu_out_var;
+}
```

Comment:
Shouldn't that part be in the file with the pass implementation?

Reply:
It was designed so in few revisions back, but author has moved some code (which I think was common in two files) to the graph_pattern_detector to be reused. I will stick to the author's design.